### PR TITLE
CLN: use standard method to set unhashable object

### PR DESCRIPTION
**Pull request by fangchenli**
#20589 mentioned lgtm. So I decided to test it. And it found this potential improvement. There is even a detailed reference with code examples. https://lgtm.com/rules/1780095/ Very interesting.

Just trying to see if their suggestion works.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -1764,11 +1764,7 @@ def _drop_labels_or_levels(self, keys, axis: int = 0):
     # ----------------------------------------------------------------------
     # Iteration
 
-    def __hash__(self):
-        raise TypeError(
-            f"{repr(type(self).__name__)} objects are mutable, "
-            f"thus they cannot be hashed"
-        )
+    __hash__ = None
```
Comment (jreback) [inline pandas/core/generic.py:1767]: can you add a comment / ref for this

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm

**Comment by jreback** [comment]
ci checks is failing as well

**Comment by fangchenli** [comment]
> ci checks is failing as well

`pandas/core/generic.py:1767: error: Incompatible types in assignment (expression has type "None", base class "object" defined the type as "Callable[[object], int]")  [assignment]`

If __hash__=None is the desired way to indicate unhashable object, this could be a mypy issue.

**Comment by fangchenli** [comment]
xref https://github.com/python/mypy/issues/4266

**Comment by fangchenli** [comment]
close for now. It doesn't work with mypy.

**Pull request closed without merging**